design/0003-capture-egress-HTTPS-metrics.md

```diff
@@ -0,0 +1,87 @@
+# Capture Egress HTTPS Metrics
+
+- @vaniisgh
+- 2020-05-16
+- Linkerd Issue: [linkerd/linkerd2#3190](https://github.com/linkerd/linkerd2/issues/3190) & [linkerd/linerd2#2192](https://github.com/linkerd/linkerd2/issues/2192)
+- Reviewers:
+
+## Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+- Outbound traffic is as important as inbound traffic, it is essential we have a
+symmetrical system to log, monitor, and provide security to external requests.
+- Often times the meshed service communicates with a third-party service suchas github.com,
+in such cases the proxy sidecars never see the unencryptedbits. This means that the
+rich metrics that Linkerd provides aren't availablefor such communications.
+- Egress traffic exist whenever a service makes requests to interacts with athird pary
+service. It is common now for such traffic to be encrypted. In suchcases, it only seems
+natural that the service mesh should be able to inspect andcontrol the flow of the same.
+- This proposal aims to provide a comprehensive solution for dealing with
+egresstraffic. Through the following sections, I aim to share my understanding of
+an effective, efficient as well as user-friendly solution that enables linkerdproxies
+to inspect engress trafic.
+- There will be a few changes that will be required on the application code,and
+modifications will be made to the way Linkerd2 deals with outbound traffic.
+
+## Design proposal (Step 2)
```

Comment:
I'd love to see a section devoted to just the requirements and tradeoffs. That'll help us flesh out the design section more.